pkg/web_app/lib/src/account.dart

```diff
@@ -516,29 +497,62 @@ class _ConsentWidget {
     if (!window.confirm('Are you sure you want to accept?')) {
       return;
     }
-    try {
+    await _rpc(() async {
       final rs = await client.resolveConsent(
           pageData.consentId, ConsentResult(granted: true));
       _updateButtons(rs.granted);
-    } on RequestException catch (e) {
-      final map = e.bodyAsJson();
-      window.alert(map['message'] as String);
-    }
+    });
   }
 
   Future _reject() async {
     if (!window.confirm('Are you sure you want to reject?')) {
       return;
     }
-    try {
+    await _rpc(() async {
       final rs = await client.resolveConsent(
           pageData.consentId, ConsentResult(granted: false));
       _updateButtons(rs.granted);
-    } on RequestException catch (e) {
-      final map = e.bodyAsJson();
-      window.alert(map['message'] as String);
-    }
+    });
   }
 
   bool get isActive => _content != null && _buttons != null;
 }
+
+Future<R> _rpc<R>(Future<R> fn()) async {
+  // Disable only the selected Elements that are not disabled at the moment.
+  final inputs = document
+      .querySelectorAll('input')
+      .cast<InputElement>()
+      .where((e) => !e.disabled)
+      .toList();
+  final buttons = document
+      .querySelectorAll('button.pub-button')
+      .cast<ButtonElement>()
+      .where((e) => !e.disabled)
+      .toList();
+
+  buttons.forEach((e) => e.disabled = true);
+  inputs.forEach((e) => e.disabled = true);
+  try {
+    return await fn();
+  } on RequestException catch (e) {
+    final message = _requestExceptionMessage(e) ?? 'Unexpected error: $e';
+    // TODO: render as a modal alert window
+    // TODO: render as HTML
+    window.alert(message);
+    rethrow;
+  } finally {
```

Comment:
Let's at-least add a TODO to handle network errors, or other errors the browser may throw... We could certainly display a generic error that we had connectivity issues :)